g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/timezone/CldrTimeZoneUtils.java

```diff
@@ -0,0 +1,149 @@
+                name.setLongStandard(stard);
+            } else {
+                String longStandard = (String) select(metazoneValue, "long.standard");
+                String longDaylight = (String) select(metazoneValue, "long.daylight");
+                String longGeneric = (String) select(metazoneValue, "long.generic");
+                String shortStandard = (String) select(metazoneValue, "short.standard");
+                String shortDaylight = (String) select(metazoneValue, "short.daylight");
+                String shortGeneric = (String) select(metazoneValue, "short.generic");
+                name.setLongStandard(longStandard);
+                name.setLongDaylight(longDaylight);
+                name.setLongGeneric(longGeneric);
+                name.setShortStandard(shortStandard);
+                name.setShortDaylight(shortDaylight);
+                name.setShortGeneric(shortGeneric);
+            }
+            metaZones.add(new CldrMetaZone(zoneKey, exemplarCity, metazoneKey, timeZone, name,
+                    territory));
+        }
+
+        TimeZoneName zone = new TimeZoneName(language, gmtZeroFormat, gmtFormat, hourFormat,
+                regionFormat, regionFormatTypeDaylight, regionFormatTypeStandard, fallbackFormat,
+                metaZones);
+        String result = null;
+        try {
+            result = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(zone);
+        } catch (JsonProcessingException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    private static String findTimeZone(String targetId, String gmtFormat, String gmtZeroFormat,
+            String hourFormat) {
+
+        TimeZone timeZone = TimeZone.getTimeZone(targetId);
+        
+
+        int offset = timeZone.getRawOffset();
+        // System.out.println(timeZone.getDisplayName()+" gmt"+offset/3600000);
+
+        if (offset > 0) {
+            Date date = new Date(offset);
+            SimpleDateFormat outputFormat = new SimpleDateFormat(hourFormat.split(";")[0].trim());
+            outputFormat.setTimeZone(TimeZone.getTimeZone(gmtZeroFormat));
+            return MessageFormat.format(gmtFormat, outputFormat.format(date));
+
+        } else if (offset < 0) {
+            Date date = new Date(0 - offset);
+            SimpleDateFormat outputFormat = new SimpleDateFormat(hourFormat.split(";")[1].trim());
+            outputFormat.setTimeZone(TimeZone.getTimeZone(gmtZeroFormat));
+            return MessageFormat.format(gmtFormat, outputFormat.format(date));
+
+        } else {
+            return gmtZeroFormat;
+        }
+    }
+
+    /**
+     * Get the node value of JSON string. e.g. main.locale.day
+     * @param jsonObj JSONObject
+     * @param keyPath JSON node path
+     * @return the node value
+     */
+    private static Object select(JSONObject jsonObj, String keyPath) {
```

Comment:
Instead of creating a select method, you can use it directly JSONUtil.select()

Reply:
done